src/adapters/IDB.js

```diff
@@ -144,38 +219,54 @@ export default class IDB extends BaseAdapter {
    * @override
    * @return {Promise}
    */
-  open() {
+  async open() {
     if (this._db) {
-      return Promise.resolve(this);
+      return this;
     }
-    return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.dbname, 1);
-      request.onupgradeneeded = event => {
-        // DB object
+
+    // In previous versions, we used to have a database with name `${bid}/${cid}`.
+    // Check if it exists, and migrate data once new schema is in place.
+    // Note: the built-in migrations from IndexedDB can only be used if the
+    // database name does not change.
+    const toMigrate = this._options.migrateOldData
+      ? await migrationRequired(this.cid)
+      : null;
+
+    this._db = await open(this.dbName, {
+      version: 1,
+      onupgradeneeded: event => {
         const db = event.target.result;
-        // Main collection store
-        const collStore = db.createObjectStore(this.storeName, {
-          keyPath: "id",
+        // Records store
+        const recordsStore = db.createObjectStore("records", {
+          keyPath: ["_cid", "id"],
         });
-        // Primary key (generated by IdSchema, UUID by default)
-        collStore.createIndex("id", "id", { unique: true });
+        // An index to obtain all the records in a collection.
+        recordsStore.createIndex("cid", "_cid");
+        // Here we create indices for every known field in records by collection.
+        // Record id (generated by IdSchema, UUID by default)
+        recordsStore.createIndex("id", ["_cid", "id"]);
```

Comment:
It's not necessary indeed. But because of the code in `createListRequest()`, it is a lot easier to manage if there is an index with the same name as the indexed column, so I left the `id` one.